ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -525,6 +521,34 @@ private TransactionWithMetadata transactionByHeaderAndIndex(
         receipts, number, query, blockhash, matchingLogs, transaction, logHasBeenRemoved);
   }
 
+  public static List<LogWithMetadata> generateLogWithMetadataForTransaction(
```

Comment:
Is this best placed in `BlockchainQueries` (unlike the other methods it is not using either of the fields of the instance) ...perhaps it could live on `TransactionRecipt` (being related to the logs of the receipt)?

Reply:
I think it living on the transaction receipt and getting the other params passed in is a good idea.

Reply:
This is consistent with the other methods in the class and how they are used.  Also, given how big this PR already is I don't think now is the time to start re-architecting.

Reply:
Also, `LogWithMetatdata` only exists in this package, moving it to `TransactionReceipt` would start leaking out implementation details.

Reply:
The PR size is a fair point. 

> LogWithMetatdata only exists in this package,
Actually, I did not spot that ...and leaking implementation is less than ideal 🤔

Reply:
The thing that jumped out at me about the method was the `static` modifier, making it the first one of the class ...but no big deal.

Reply:
It is truly a helper method.  `LogWithMetadata` could be a good home but like the other classes currently it is a pure data object class so I wanted to preserve it for the time being.